pywinauto/application.py

```diff
@@ -587,7 +587,7 @@ def print_control_identifiers(self, depth=None, filename=None):
         log_file = None
         if filename is not None:
             log_file = open(filename, "w")
-            log_func = log_file.write
+            log_func = lambda arg: log_file.write(str(arg) + os.linesep)

```

Comment:
I've already tested variant with `def`. It causes a critical warning related to redefinition of `log_func` identifier (it had been already defined few lines upper). I'll move `if` which resolves log_finc after `def print_identifiers` to avoid this problem.